### PR TITLE
Add root-config.bat for Windows

### PR DESCRIPTION
**Pull request by bellenot**
Add an (almost) `root-config` equivalent script for Windows, but unfortunately not usable to compile code in a terminal, since command substitution doesn't work in the Windows command prompt...

### CODE REVIEW

**Review by pcanal — APPROVED** [review]

File: config/root-config.bat.in [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+   " [--f77] [--ld ] [--help]"
+) do set usage=!usage!%%~u^
+
+
+if "%1"=="" (
+   echo !usage! 1>&2
+   exit /b 1
+)
+
+set arch=@architecture@
```
[the remark this answers is not in the document]
Comment (bellenot) [reply config/root-config.bat.in:24]: Not sure I understand... `arch` is the architecture used to build ROOT (i.e. taken from CMake), which is the relevant info. On Windows, even when building ROOT in 32 bit mode in the command prompt (which is 64 bit), what defines the architecture (32/64) is which MSVC batch script you call (or in which environment you are)

[the remark this answers is not in the document]
Comment (bellenot) [reply config/root-config.bat.in:24]: And BTW, I simply took the same value than the original `root-config` script...

[the remark this answers is not in the document]
Comment (pcanal) [reply config/root-config.bat.in:24]: 'arch' was just an example.  Replace it with cxxflags to make it more useful.

What I was thinking about is "vcvar32.bat" leaves behind environment variable and we might be able to do the same.  So the user could then do
```cl %root_cxxflags %root_ldflags .... ```

[the remark this answers is not in the document]
Comment (bellenot) [reply config/root-config.bat.in:24]: > What I was thinking about is "vcvar32.bat" leaves behind environment variable and we might be able to do the same. So the user could then do
> `cl %root_cxxflags %root_ldflags .... `

No, it doesn't, the variables are local and don't "pollute" the environment variables of the caller

[the remark this answers is not in the document]
Comment (bellenot) [reply config/root-config.bat.in:24]: But I could add some options to do so.... I'll check what can be done

[the remark this answers is not in the document]
Comment (bellenot) [reply config/root-config.bat.in:24]: @pcanal OK, so I tried and using environment variables doesn't work, it's still the problem of the command substitution which is not working in the command prompt. So I'll merge this PR. Thanks for the comments and the suggestion!

**Pull request merged**